chain/client/src/stateless_validation/validate.rs

```diff
@@ -92,60 +113,64 @@ pub fn validate_chunk_endorsement(
     epoch_manager: &dyn EpochManagerAdapter,
     endorsement: &ChunkEndorsement,
     store: &Store,
-) -> Result<bool, Error> {
-    if !validate_chunk_relevant_as_validator(
+) -> Result<ChunkRelevance, Error> {
+    match validate_chunk_relevant_as_validator(
         epoch_manager,
         &endorsement.chunk_production_key(),
         endorsement.account_id(),
         store,
     )? {
-        return Ok(false);
+        ChunkRelevance::Relevant => {}
+        irrelevant => return Ok(irrelevant),
     }
     validate_chunk_endorsement_signature(epoch_manager, endorsement)?;
 
-    Ok(true)
+    Ok(ChunkRelevance::Relevant)
 }
 
 pub fn validate_chunk_contract_accesses(
     epoch_manager: &dyn EpochManagerAdapter,
     accesses: &ChunkContractAccesses,
     signer: &ValidatorSigner,
     store: &Store,
-) -> Result<bool, Error> {
+) -> Result<ChunkRelevance, Error> {
     let key = accesses.chunk_production_key();
-    if !validate_chunk_relevant_as_validator(epoch_manager, key, signer.validator_id(), store)? {
-        return Ok(false);
+    match validate_chunk_relevant_as_validator(epoch_manager, key, signer.validator_id(), store)? {
```

Comment:
nit: In plenty of places the method names refer to chunk instead of chunk_endorsement - can you rename those? Can be in a separate PR. 

nit2: Maybe consider something like check_chunk_endorsement_relevance? It's nice to have the return type name in the method (ChunkEndorsementRelevance -> check_**chunk_endorsement_relevance**)

nit3: This pattern repeating a few times is a bit strange, is there any nice refactoring to do it neater? The immediate thought is to use the `?` but it's already used for errors. Is there any other smart way about it?

Reply:
ad. 1 & 2: See my comment about partial witness use case above.

ad. 3: I've tried using nested `Results`, but it was kind of ugly. I can add a custom macro for this, but not sure if it's worth the effort.

Reply:
Did the macro after all. Wasn't that much of an effort. 😛